ufo2fontir/src/source.rs

```diff
@@ -845,6 +863,79 @@ impl Work<Context, WorkError> for FeatureWork {
     }
 }
 
+impl Work<Context, WorkError> for KerningWork {
+    fn exec(&self, context: &Context) -> Result<(), WorkError> {
+        debug!("Kerning for {:#?}", self.designspace_file);
+
+        let designspace_dir = self.designspace_file.parent().unwrap();
+        let static_metadata = context.get_final_static_metadata();
+        let master_locations = master_locations(&static_metadata.axes, &self.designspace.sources);
+
+        let mut kerning = Kerning::default();
+        for source in self.designspace.sources.iter() {
+            let pos = master_locations.get(&source.name).unwrap();
+            let ufo_dir = designspace_dir.join(&source.filename);
+            let data_request = norad::DataRequest::none().groups(true).kerning(true);
+            let font = norad::Font::load_requested_data(&ufo_dir, data_request)
+                .map_err(|e| WorkError::ParseError(ufo_dir, format!("{e}")))?;
+
+            kerning.groups = font
+                .groups
+                .into_iter()
+                .map(|(group_name, entries)| {
+                    (
+                        GroupName::from(group_name.as_str()),
+                        entries
+                            .into_iter()
+                            .filter_map(|glyph_name| {
+                                let glyph_name = GlyphName::from(glyph_name.as_str());
+                                if static_metadata.glyph_order.contains(&glyph_name) {
+                                    Some(glyph_name)
+                                } else {
+                                    warn!("{} kerning group {} references non-existent glyph {}; ignoring", source.filename, group_name, glyph_name);
+                                    None
+                                }
+                            })
+                            .collect(),
+                    )
+                })
+                .collect();
+
+            let resolve_participant = |name: &norad::Name| {
+                let group_name = GroupName::from(name.as_str());
+                if kerning.groups.contains_key(&group_name) {
```

Comment:
it's not the mere fact that the name of a kerning pair's member matches the name of one of the UFO groups that makes it a "kerning" group. You don't want to accidentally consider a kerning participant named "Rod" to be a "group" just because there's a group inside groups.plist that's named like that.
There's the public.kern{1,2}. prefix I mentioned above that has special meaning/restrictions (kern1 can only occur in first logical position, kern2 only on the second).
You should take a look at how ufo2ft's KernFeatureWriter collects UFO's kerning groups and pairs:
https://github.com/googlefonts/ufo2ft/blob/3e0563814cf541f7d8ca2bb7f6e446328e0e5e76/Lib/ufo2ft/featureWriters/kernFeatureWriter.py#L302-L357

Reply:
Good catch, fixed, ty.